### PR TITLE
Added OnOpen and OnClose params for container

### PR DESCRIPTION
**Pull request by jopnine**
Added a call for OnOpen and OnClose in order to make containers interactions more customizable.

### CODE REVIEW

File: plugins/containers/entities/entities/ix_container.lua [hunk 1]
```diff
@@ -73,14 +73,23 @@ if (SERVER) then
 
 		if (inventory) then
 			local name = self:GetDisplayName()
-
+			local def = ix.container.stored[self:GetModel():lower()]
```
Comment (garytate) [inline plugins/containers/entities/entities/ix_container.lua:76]:
```suggestion
			local definition = ix.container.stored[self:GetModel():lower()]
			
```

File: plugins/containers/entities/entities/ix_container.lua [hunk 2]
```diff
@@ -73,14 +73,23 @@ if (SERVER) then
...
 			ix.storage.Open(activator, inventory, {
 				name = name,
 				entity = self,
+				bMultipleUsers = true,
```
Comment (garytate) [inline plugins/containers/entities/entities/ix_container.lua:80]: Why has this been added?

File: plugins/containers/entities/entities/ix_container.lua [hunk 3]
```diff
@@ -73,14 +73,23 @@ if (SERVER) then
...
 				searchTime = ix.config.Get("containerOpenTime", 0.7),
 				data = {money = self:GetMoney()},
+				OnPlayerOpen = function()
+					if def.OnOpen then
+						def.OnOpen(self, activator)
+					end
+				end,
 				OnPlayerClose = function()
 					ix.log.Add(activator, "closeContainer", name, inventory:GetID())
```
Comment (garytate) [inline plugins/containers/entities/entities/ix_container.lua:89]:
```suggestion
					if (definition.OnClose) then
                        definition.OnClose(self, activator)
                    end
                    
					ix.log.Add(activator, "closeContainer", name, inventory:GetID())
```

File: plugins/containers/entities/entities/ix_container.lua [hunk 4]
```diff
@@ -73,14 +73,23 @@ if (SERVER) then
...
-
+			local def = ix.container.stored[self:GetModel():lower()]
 			ix.storage.Open(activator, inventory, {
 				name = name,
 				entity = self,
+				bMultipleUsers = true,
 				searchTime = ix.config.Get("containerOpenTime", 0.7),
 				data = {money = self:GetMoney()},
+				OnPlayerOpen = function()
+					if def.OnOpen then
```
Comment (garytate) [inline plugins/containers/entities/entities/ix_container.lua:84]:
```suggestion
					if (definition.OnOpen) then
					    definition.OnOpen(self, activator)
					end
```

**Review by garytate — CHANGES_REQUESTED** [review]

**Review by alexgrist — APPROVED** [review]

**Pull request merged**